### PR TITLE
Nats support

### PR DESCRIPTION
**Pull request by YuliaShvets**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -28,13 +28,16 @@ dependencies {
     implementation("org.springframework.boot:spring-boot-starter-web")
     implementation("com.fasterxml.jackson.module:jackson-module-kotlin")
     implementation("org.jetbrains.kotlin:kotlin-reflect")
+    implementation(project(mapOf("path" to ":nats")))
```
Comment (ajax-gnatiuk-s) [inline build.gradle.kts]:
```suggestion
    implementation(project(":nats"))
```

File: nats/.gitignore [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+### NetBeans ###
+/nbproject/private/
+/nbbuild/
+/dist/
+/nbdist/
+/.nb-gradle/
+
+### VS Code ###
+.vscode/
```
Comment (ajax-gnatiuk-s) [inline nats/.gitignore]: why do we need the separate gitignore file?

File: nats/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -0,0 +1,7 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.2.1-bin.zip
+networkTimeout=10000
+validateDistributionUrl=true
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
```
Comment (ajax-gnatiuk-s) [inline nats/gradle/wrapper/gradle-wrapper.properties]: we don't need one more copy of gradle wrapper for the subproject.

File: src/main/java/ua/lviv/iot/parkingServer/converter/ParkingConverter.kt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+@Component
+class ParkingConverter {
+    fun parkingToProto(parking: Parking): ParkingOuterClass.Parking {
+        return ParkingOuterClass.Parking.newBuilder()
+            .setLocation(parking.location)
+            .setTradeNetwork(parking.tradeNetwork)
+            .setCountOfParkingSpots(parking.countOfParkingSpots)
+            .build()
+
```
Comment (ajax-gnatiuk-s) [inline src/main/java/ua/lviv/iot/parkingServer/converter/ParkingConverter.kt:19]:
```suggestion
```

File: src/main/java/ua/lviv/iot/parkingServer/natsconfig/NatsConfig.kt [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+class NatsConfig {
+    @Value("\${nats.connection.url}")
+    private lateinit var natsUrl: String
+
+    @Bean
+    fun natsConnection(): Connection = Nats.connect(natsUrl)
```
Comment (ajax-gnatiuk-s) [inline src/main/java/ua/lviv/iot/parkingServer/natsconfig/NatsConfig.kt]:
```suggestion

    @Bean
    fun natsConnection(
        @Value("\${nats.connection.url}") natsUrl: String
    ): Connection = Nats.connect(natsUrl)
```

File: src/main/java/ua/lviv/iot/parkingServer/natsconfig/NatsConfigurationBeanPostProcessor.kt [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package ua.lviv.iot.parkingServer.natsconfig
+
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.stereotype.Component
+import ua.lviv.iot.parkingServer.natscontroller.NatsController
+
+@Component
+class  NatsConfigurationBeanPostProcessor : BeanPostProcessor {
+    override fun postProcessBeforeInitialization(bean: Any, beanName: String): Any {
```
Comment (ajax-gnatiuk-s) [inline src/main/java/ua/lviv/iot/parkingServer/natsconfig/NatsConfigurationBeanPostProcessor.kt]:
```suggestion
class  NatsConfigurationBeanPostProcessor : BeanPostProcessor {
    
    override fun postProcessBeforeInitialization(bean: Any, beanName: String): Any {
```

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsController.kt [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+interface NatsController<ReqT : GeneratedMessageV3, RespT : GeneratedMessageV3> {
+    val subject: String
+    val connection: Connection
+    val parser: Parser<ReqT>
+    fun generateReplyForNatsRequest(request: ReqT): RespT
+    fun handle(msg: Message): RespT {
+        val request: ReqT = parser.parseFrom(msg.data)
+        return generateReplyForNatsRequest(request)
+    }
```
Comment (ajax-gnatiuk-s) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsController.kt]:
```suggestion
interface NatsController<ReqT : GeneratedMessageV3, RespT : GeneratedMessageV3> {

    val subject: String
    
    val connection: Connection
    
    val parser: Parser<ReqT>
    
    fun generateReplyForNatsRequest(request: ReqT): RespT
    
    fun handle(msg: Message): RespT {
        val request: ReqT = parser.parseFrom(msg.data)
        return generateReplyForNatsRequest(request)
    }
```

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingCreateController.kt [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import ua.lviv.iot.parkingServer.service.interfaces.ParkingServiceInterface
+
+@Component
+class NatsParkingCreateController(
+    private val converter: ParkingConverter,
+    private val service: ParkingServiceInterface,
+    override val connection: Connection
+) : NatsController<ParkingRequest, ParkingResponse> {
+    override val subject: String = "parking.add"
+    override val parser: Parser<ParkingRequest> =
```
Comment (ajax-gnatiuk-s) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingCreateController.kt]:
```suggestion
) : NatsController<ParkingRequest, ParkingResponse> {

    override val subject: String = "parking.add"
    
    override val parser: Parser<ParkingRequest> =
```

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteController.kt [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+import ua.lviv.iot.parkingServer.service.interfaces.ParkingServiceInterface
+
+@Component
+class NatsParkingDeleteController(
+    private val converter: ParkingConverter,
+    private val service: ParkingServiceInterface,
+    override val connection: Connection,
+) : NatsController<ParkingRequest, ParkingResponse> {
+    override val subject: String = "parking.delete"
+    override val parser: Parser<ParkingRequest> = ParkingRequest.parser()
```
Comment (ajax-gnatiuk-s) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteController.kt]:
```suggestion
) : NatsController<ParkingRequest, ParkingResponse> {

    override val subject: String = "parking.delete"
    
    override val parser: Parser<ParkingRequest> = ParkingRequest.parser()
```

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingGetByIdController.kt [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+import ua.lviv.iot.parkingServer.service.interfaces.ParkingServiceInterface
+
+@Component
+class NatsParkingGetByIdController(
+    private val converter: ParkingConverter,
+    private val service: ParkingServiceInterface,
+    override val connection: Connection,
+) : NatsController<GetByIdParkingRequest, GetByIdParkingResponse> {
+    override val subject: String = "parking.get_by_id"
+    override val parser: Parser<GetByIdParkingRequest> = GetByIdParkingRequest.parser()
```
Comment (ajax-gnatiuk-s) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingGetByIdController.kt]:
```suggestion
) : NatsController<GetByIdParkingRequest, GetByIdParkingResponse> {

    override val subject: String = "parking.get_by_id"
    
    override val parser: Parser<GetByIdParkingRequest> = GetByIdParkingRequest.parser()
```

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+import ua.lviv.iot.parkingServer.service.interfaces.ParkingServiceInterface
+
+@Component
+class NatsParkingUpdateController(
+    private val converter: ParkingConverter,
+    private val service: ParkingServiceInterface,
+    override val connection: Connection,
+) : NatsController<ParkingOuterClass.ParkingRequest, ParkingOuterClass.ParkingResponse> {
+    override val subject: String = "parking.update"
+    override val parser: Parser<ParkingOuterClass.ParkingRequest> = ParkingOuterClass.ParkingRequest.parser()
```
Comment (ajax-gnatiuk-s) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt]:
```suggestion
) : NatsController<ParkingOuterClass.ParkingRequest, ParkingOuterClass.ParkingResponse> {

    override val subject: String = "parking.update"
    
    override val parser: Parser<ParkingOuterClass.ParkingRequest> = ParkingOuterClass.ParkingRequest.parser()
```

File: src/main/resources/application-docker.properties [hunk 12]
```diff
@@ -0,0 +1,2 @@
+spring.data.mongodb.host=mongo
+nats.connection.url=nats://nats:4222
```
Comment (ajax-gnatiuk-s) [inline src/main/resources/application-docker.properties:2]: :+1:

File: src/main/resources/application.properties [hunk 13]
```diff
@@ -1,6 +1,4 @@
-spring.data.mongodb.authentication-database=admin
-spring.data.mongodb.username=root
-spring.data.mongodb.password=password
 spring.data.mongodb.database=parking-db
+spring.data.mongodb.host=localhost
 spring.data.mongodb.port=27017
-spring.data.mongodb.host=mongo
+nats.connection.url=nats://localhost:4222
```
Comment (ajax-gnatiuk-s) [inline src/main/resources/application.properties:4]: :+1:

File: src/test/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingCreateControllerTest.kt [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+            connection
+                .requestWithTimeout(
+                    "parking.add",
+                    expected.toByteArray(),
+                    Duration.ofMillis(100000)
+                )
+                .get()
+                .data
+        )
+        assertThat(expected.parking).isEqualTo(actual.parking)
```
Comment (ajax-gnatiuk-s) [inline src/test/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingCreateControllerTest.kt]:
```
assertThat(<ACTUAL>).isEqualTo(<EXPECTED>)
```

Compare two cases:
1. :-1: 
```
val expected = 1
val actual = 2
assertThat(expected).isEqualTo(actual)

Error msg generated by the assertion framework:
expected:<[2]> but was:<[1]>
```

2. :+1: 
```
val expected = 1 
val actual = 2
assertThat(actual).isEqualTo(expected)

Error msg generated by the assertion framework:
expected:<[1]> but was:<[2]>
```

File: src/test/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteControllerTest.kt [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+    fun generateReplyForNatsRequest() {
+        val parking = Parking("Kyiv", "Forum", 123)
+        parkingRepository.save(parking)
+        val sizeOfDBBefore = parkingRepository.findAll().size
+        val future = connection.requestWithTimeout(
+            "parking.delete",
+            null,
+            Duration.ofMillis(100000)
+        )
+        future.get().data
```
Comment (ajax-gnatiuk-s) [inline src/test/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteControllerTest.kt]:
```suggestion
        connection.requestWithTimeout(
            "parking.delete",
            null,
            Duration.ofSeconds(10)
        ).get()
```

**Review by ajax-gnatiuk-s — CHANGES_REQUESTED** [review]

**Comment by ajax-gnatiuk-s** [comment]
@tetty-pa please review

File: build.gradle.kts (hunk 1 above)
Comment (YuliaShvets) [reply build.gradle.kts]: Fixed

File: nats/.gitignore (hunk 2 above)
Comment (YuliaShvets) [reply nats/.gitignore]: Committed by accident

File: nats/gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (YuliaShvets) [reply nats/gradle/wrapper/gradle-wrapper.properties]: Fixed

File: src/main/java/ua/lviv/iot/parkingServer/converter/ParkingConverter.kt (hunk 4 above)
Comment (YuliaShvets) [reply src/main/java/ua/lviv/iot/parkingServer/converter/ParkingConverter.kt:19]: ✅

File: src/main/java/ua/lviv/iot/parkingServer/natsconfig/NatsConfig.kt (hunk 5 above)
Comment (YuliaShvets) [reply src/main/java/ua/lviv/iot/parkingServer/natsconfig/NatsConfig.kt]: ✅

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt (hunk 11 above)
Comment (YuliaShvets) [reply src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt]: ✅

File: src/test/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingCreateControllerTest.kt (hunk 14 above)
Comment (YuliaShvets) [reply src/test/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingCreateControllerTest.kt]: ✅

File: src/test/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteControllerTest.kt (hunk 15 above)
Comment (YuliaShvets) [reply src/test/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteControllerTest.kt]: ✅

File: build.gradle.kts [hunk 16]
```diff
@@ -28,13 +28,16 @@ dependencies {
     implementation("org.springframework.boot:spring-boot-starter-web")
     implementation("com.fasterxml.jackson.module:jackson-module-kotlin")
     implementation("org.jetbrains.kotlin:kotlin-reflect")
```
Comment (tetty-pa) [inline build.gradle.kts:33]: maybe it's better to sort and group your dependencies)

File: nats/build.gradle.kts [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+dependencies {
+    implementation("com.google.protobuf:protobuf-java:3.24.2")
+    implementation("com.google.protobuf:protobuf-java-util:3.20.1")
+}
+
+kotlin {
+    jvmToolchain(17)
+}
+
+
```
Comment (tetty-pa) [inline nats/build.gradle.kts]: empty line

File: src/main/java/ua/lviv/iot/parkingServer/converter/ParkingConverter.kt [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+import ua.lviv.iot.parkingServer.model.Parking
+
+@Component
+class ParkingConverter {
+    fun parkingToProto(parking: Parking): ParkingOuterClass.Parking {
+        return ParkingOuterClass.Parking.newBuilder()
+            .setLocation(parking.location)
+            .setTradeNetwork(parking.tradeNetwork)
+            .setCountOfParkingSpots(parking.countOfParkingSpots)
+            .build()
```
Comment (tetty-pa) [inline src/main/java/ua/lviv/iot/parkingServer/converter/ParkingConverter.kt]:
```suggestion
        return ParkingOuterClass.Parking.newBuilder().apply{
            location = parking.location
            tradeNetwork = parking.tradeNetwork
            countOfParkingSpots = parking.countOfParkingSpots
         }.build()
```

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingCreateController.kt [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+import ua.lviv.iot.parkingServer.service.interfaces.ParkingServiceInterface
+
+@Component
+class NatsParkingCreateController(
+    private val converter: ParkingConverter,
+    private val service: ParkingServiceInterface,
+    override val connection: Connection
+) : NatsController<ParkingRequest, ParkingResponse> {
+
+    override val subject: String = "parking.add"
```
Comment (tetty-pa) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingCreateController.kt]: You can separate your subjects as constants in one class in nats module

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteController.kt [hunk 20]
```diff
@@ -0,0 +1,30 @@
...
+
+    override val parser: Parser<ParkingRequest> = ParkingRequest.parser()
+
+    override fun generateReplyForNatsRequest(request: ParkingRequest): ParkingResponse {
+        val parking: Parking = converter.protoRequestToParking(request)
+        parking.id = subject
+        service.deleteEntity(subject)
+        return ParkingResponse.newBuilder().build()
+    }
+
```
Comment (tetty-pa) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteController.kt]: empty line

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingGetByIdController.kt [hunk 21]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.stereotype.Component
+import ua.lviv.iot.parkingServer.converter.ParkingConverter
+import ua.lviv.iot.parkingServer.service.interfaces.ParkingServiceInterface
+
+@Component
+class NatsParkingGetByIdController(
+    private val converter: ParkingConverter,
+    private val service: ParkingServiceInterface,
+    override val connection: Connection,
+) : NatsController<GetByIdParkingRequest, GetByIdParkingResponse> {
```
Comment (tetty-pa) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingGetByIdController.kt:17]:
```suggestion
): NatsController<GetByIdParkingRequest, GetByIdParkingResponse> {
```

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteController.kt [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+import ua.lviv.iot.parkingServer.converter.ParkingConverter
+import ua.lviv.iot.parkingServer.model.Parking
+import ua.lviv.iot.parkingServer.service.interfaces.ParkingServiceInterface
+
+@Component
+class NatsParkingDeleteController(
+    private val converter: ParkingConverter,
+    private val service: ParkingServiceInterface,
+    override val connection: Connection,
+) : NatsController<ParkingRequest, ParkingResponse> {
```
Comment (tetty-pa) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteController.kt]:
```suggestion
): NatsController<ParkingRequest, ParkingResponse> {
```

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingGetByIdController.kt (hunk 21 above)
Comment (tetty-pa) [reply src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingGetByIdController.kt:17]: fix this in all classes

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt [hunk 23]
```diff
@@ -0,0 +1,28 @@
...
+    override val connection: Connection,
+) : NatsController<ParkingOuterClass.ParkingRequest, ParkingOuterClass.ParkingResponse> {
+
+    override val subject: String = "parking.update"
+
+    override val parser: Parser<ParkingOuterClass.ParkingRequest> = ParkingOuterClass.ParkingRequest.parser()
+
+    override fun generateReplyForNatsRequest(request: ParkingOuterClass.ParkingRequest): ParkingOuterClass.ParkingResponse {
+        val parking: Parking = converter.protoRequestToParking(request)
+        parking.id = subject
```
Comment (tetty-pa) [inline src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt]: why `parking.id = subject`?

File: src/test/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingCreateControllerTest.kt [hunk 24]
```diff
@@ -0,0 +1,42 @@
+package ua.lviv.iot.parkingServer.natscontroller
+
+import com.example.ParkingOuterClass
+import io.nats.client.Connection
+import java.time.Duration
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+
```
Comment (tetty-pa) [inline src/test/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingCreateControllerTest.kt]: empty line

File: src/test/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteControllerTest.kt [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
+        connection.requestWithTimeout(
+            "parking.delete",
+            null,
+            Duration.ofSeconds(10)
+        ).get().data
+
+        val sizeOfDBAfter = parkingRepository.findAll().size
+
+        assertThat(sizeOfDBBefore).isEqualTo(sizeOfDBAfter)
+
```
Comment (tetty-pa) [inline src/test/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteControllerTest.kt]: empty line

File: src/test/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateControllerTest.kt [hunk 26]
```diff
@@ -0,0 +1,41 @@
...
+        val future = connection.requestWithTimeout(
+            "parking.update",
+            expected.toByteArray(),
+            Duration.ofSeconds(10)
+        )
+            .get().data
+
+        val actual = ParkingOuterClass.ParkingResponse.parseFrom(future)
+        assertThat(actual.parking).isEqualTo(expected.parking)
+
```
Comment (tetty-pa) [inline src/test/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateControllerTest.kt:48]: empty line

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteController.kt (hunk 22 above)
Comment (ajax-gnatiuk-s) [reply src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingDeleteController.kt]: No, current code style is correct

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingGetByIdController.kt (hunk 21 above)
Comment (ajax-gnatiuk-s) [reply src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingGetByIdController.kt:17]: please, don't touch. Everything is ok here

File: src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt (hunk 23 above)
Comment (YuliaShvets) [reply src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingUpdateController.kt]: TBH, I'm little confused what I've written here :(
Already fixed and works correctly

File: build.gradle.kts [hunk 27]
```diff
@@ -26,15 +26,18 @@ repositories {
...
     compileOnly("org.projectlombok:lombok")
-    developmentOnly("org.springframework.boot:spring-boot-devtools")
     annotationProcessor("org.projectlombok:lombok")
-    testImplementation("org.springframework.boot:spring-boot-starter-test")
     implementation(kotlin("stdlib-jdk8"))
-    implementation ("io.github.microutils:kotlin-logging-jvm:2.0.11")
-    implementation("org.springframework.boot:spring-boot-starter-data-mongodb")
+    implementation("io.nats:jnats:2.16.14")
+    implementation("com.google.protobuf:protobuf-java:3.19.6")
+    implementation(project(":nats"))
```
Comment (tetty-pa) [inline build.gradle.kts]: now sort by type of dependency (implementation, testImplementation)

Comment (YuliaShvets) [reply build.gradle.kts]: ✅

File: build.gradle.kts [hunk 28]
```diff
@@ -25,19 +25,21 @@ repositories {
...
-    compileOnly("org.projectlombok:lombok")
-    annotationProcessor("org.projectlombok:lombok")
     implementation(kotlin("stdlib-jdk8"))
     implementation("io.nats:jnats:2.16.14")
     implementation("com.google.protobuf:protobuf-java:3.19.6")
-    implementation(project(":nats"))
+
+    developmentOnly("org.springframework.boot:spring-boot-devtools")
+    compileOnly("org.projectlombok:lombok")
+    annotationProcessor("org.projectlombok:lombok")
```
Comment (ajax-gnatiuk-s) [inline build.gradle.kts]: we don't need lombok in kotlin project. Please, remove

**Review by ajax-gnatiuk-s — APPROVED** [review]

**Review by tetty-pa — APPROVED** [review]

File: build.gradle.kts (hunk 28 above)
Comment (YuliaShvets) [reply build.gradle.kts]: ✅

**Pull request merged**